addons/payment/models/account_payment.py

```diff
@@ -12,27 +12,43 @@ class AccountPayment(models.Model):
     payment_transaction_id = fields.Many2one('payment.transaction', string='Payment Transaction', readonly=True)
     payment_token_id = fields.Many2one(
         'payment.token', string="Saved payment token",
-        domain="[('acquirer_id.capture_manually', '=', False), ('company_id', '=', company_id)]",
+        domain="""[
+            ('company_id', '=', company_id),
+            ('acquirer_id.capture_manually', '=', False),
+            '|', '|', '|',
+                (payment_method_code != 'electronic', '=', 1),
+                (not partner_id, '=' 1),
+                (not journal_id, '=', 1),
+                '&', ('acquirer_id.journal_id', '=', journal_id),
+                     ('partner_id', 'in', related_partner_ids),
+        ]""",
```

Comment:
Does it really make sense to be able to select any token whatsoever if the payment_method is not electronic, or the partner is not set, or the journal is not set?

Reply:
Seems logical to smetl